### ISSUE TITLE
MSFT_SPFarmAdministrators MembersToExclude and MembersToInclude parameters are case-sensitive

### ISSUE THREAD

**Issue by egor-yudkin** [PowerShell/SharePointDsc#768]
**Details of the scenario you try and problem that is occurring:**

I was trying to use SPFarmAdministrators resource to remove built-in Administrators group from farm admins. And I was not able to do it because I was copying the group name from the Farm Administrators list. But when I changed the "A"->"a" in "BUILTIN\Administrators" parameter of configuration script, it began to work. This is what I had from the verbose output before I changed the case of "A":

> VERBOSE: [DFWDW0APP219]: LCM:  [ Start  Resource ]  [[SPFarmAdministrators]LocalFarmAdmins]
> VERBOSE: [DFWDW0APP219]: LCM:  [ Start  Test     ]  [[SPFarmAdministrators]LocalFarmAdmins]
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Testing Farm Administrators configuration
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Getting Farm Administrators configuration
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Executing as the local run as user ENT\svc_spodrsetup_d
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving BeginProcessing Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving ProcessRecord Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving EndProcessing Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving BeginProcessing Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving ProcessRecord Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving EndProcessing Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Processing MembersToExclude parameter
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] BUILTIN\Administrators is a Farm Administrator. Set result to false
> VERBOSE: [DFWDW0APP219]: LCM:  [ End    Test     ]  [[SPFarmAdministrators]LocalFarmAdmins]  in 0.9840 seconds.
> VERBOSE: [DFWDW0APP219]: LCM:  [ Start  Set      ]  [[SPFarmAdministrators]LocalFarmAdmins]
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Setting Farm Administrators configuration
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Getting Farm Administrators configuration
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Executing as the local run as user ENT\svc_spodrsetup_d
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving BeginProcessing Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving ProcessRecord Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving EndProcessing Method of Get-SPwebapplication.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving BeginProcessing Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving ProcessRecord Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Leaving EndProcessing Method of Get-SPweb.
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] Processing MembersToExclude parameter
> VERBOSE: [DFWDW0APP219]:                            [[SPFarmAdministrators]LocalFarmAdmins] BUILTIN\Administrators is not a Farm Administrator. Skipping
> VERBOSE: [DFWDW0APP219]: LCM:  [ End    Set      ]  [[SPFarmAdministrators]LocalFarmAdmins]  in 0.1720 seconds.
> VERBOSE: [DFWDW0APP219]: LCM:  [ End    Resource ]  [[SPFarmAdministrators]LocalFarmAdmins]
> 

Test-TargetResource  and Set-TargetResource functions have different opinion on the question if this group is a member of Farm Administrators. This is because different methods are used to check this.

```
function Set-TargetResource
{
...
if ($CurrentValues.Members.Contains($member))
...
}

function Test-TargetResource
{
...
if ($CurrentValues.Members -contains $member)
...
}
```

-contains is case-insensitive, but .Contains() is case-sensitive. So the resource behavior is inconsistent. I'm not sure what should be the right approach, but I believe case-sensitivity is not common in PowerShell scripts.

**The DSC configuration that is using the resource:**
N/A

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Windows Server 2012 R2, PowerShell 5.1

**Version of SharePoint that is used (e.g. SharePoint 2016):**
SharePoint 2016

**Version of the DSC module you're using:**
2.0.0.0

**Comment by ykuijs** [comment]
Great catch! They all have to be case insensitive.

Will fix this in the bugfix PR I am working on

**Comment by ykuijs** [comment]
Closed via PR #775
